src/main/java/net/sacredlabyrinth/phaed/simpleclans/commands/staff/StaffCommands.java

```diff
@@ -326,7 +327,15 @@ public void demote(CommandSender sender, @Conditions("clan_member") @Name("leade
     @Description("{@@command.description.resetkdr.everyone}")
     public void resetKdr(CommandSender sender) {
         for (ClanPlayer cp : cm.getAllClanPlayers()) {
-            cm.resetKdr(cp);
+            Player player = cp.toPlayer();
+            if (player == null) {
+                continue;
+            }
```

Comment:
The previous KDR command was clearing KDR from all clan players, even if they are offline.

Reply:
> @Tomut0 What is the best solution in this case?

IMO, PlayerResetKdrEvent could have `clanplayer` in the constructor args instead of `player`.